test/Microsoft.AspNetCore.Server.KestrelTests/BadHttpRequestTests.cs

```diff
@@ -1,72 +1,99 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Threading.Tasks;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.KestrelTests
 {
     public class BadHttpRequestTests
     {
-        [Theory(Skip = "This test fails intermittently and needs to be investigated.")]
-        [InlineData("/ HTTP/1.1\r\n\r\n")]
-        [InlineData(" / HTTP/1.1\r\n\r\n")]
-        [InlineData("  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  /  HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  HTTP/1.1\r\n\r\n")]
+        // Don't send more data than necessary to fail, otherwise the test might throw if
+        // it's still trying to send characters when the server had already shut down the connection
```

Comment:
This could also be a bug in our `TestConnection` class rather than `Socket`.  Perhaps it shouldn't call `_socket.Shutdown()` until after the response has been read.  Maybe this works in normal cases, but fails if the server closed the connection.